8/SiloHost/src/SiloConfigurationHelper.cs

```diff
@@ -0,0 +1,127 @@
+    {
+        private IEnvironmentVariables _environmentVariables;
+        public SiloConfigurationHelper(IEnvironmentVariables environmentVariables)
+        {
+            _environmentVariables = environmentVariables;
+        }
+         public void ConfigureEndpointOptions(EndpointOptions endpointOptions)
+        {
+            if (_environmentVariables.IsLocal())
+                LocalEndpointSettings(endpointOptions);
+            else
+                ElasticContainerServiceEndpointSettings(endpointOptions, _environmentVariables.EcsContainerMetadataUri());
+            
+            endpointOptions.SiloListeningEndpoint = new IPEndPoint(IPAddress.Any, 2000);
+            endpointOptions.GatewayListeningEndpoint = new IPEndPoint(IPAddress.Any, 3000);
+        }
+
+         public void ConfigureDynamoClusterOptions(DynamoDBClusteringOptions clusteringOptions)
+         {
+             clusteringOptions.TableName = _environmentVariables.MembershipTable();
+             clusteringOptions.Service = _environmentVariables.AwsRegion();
+         }
+         
+         public void ConfigureClusterOptions(ClusterOptions clusterOptions)
+         {
+             clusterOptions.ClusterId = "cluster-of-silos";
+             clusterOptions.ServiceId = "hello-world-service";
+         }
+         
+         public void ConfigureDashboardOptions(DashboardOptions dashboardOptions)
+         {
+             dashboardOptions.Username = "piotr";
+             dashboardOptions.Password = "orleans";
+             dashboardOptions.Port = _environmentVariables.DashboardPort();
+         }
+         
+        private void ElasticContainerServiceEndpointSettings(EndpointOptions endpointOptions, string ecsContainerMetadataUri)
+        {
+            var responseBody = string.Empty;
+
+            if (string.IsNullOrEmpty(ecsContainerMetadataUri))
+            {
+                throw new Exception("ECS container metadata URI cannot be null.");
+            }
+            
+            var response = new HttpClient().GetAsync(ecsContainerMetadataUri).Result;
+
+            if (response.StatusCode == HttpStatusCode.OK)
+            {
+                responseBody = response.Content.ReadAsStringAsync().Result;
+            }
+            var ecsMetadata = JsonSerializer.Deserialize<EcsMetadata>(responseBody);
+            var ip = EC2InstanceMetadata.PrivateIpAddress;
+            var siloPort = ecsMetadata?.Ports?.FirstOrDefault(x => x.ContainerPort == 2000)?.HostPort ?? 0;
+            var gatewayPort = ecsMetadata?.Ports?.FirstOrDefault(x => x.ContainerPort == 3000)?.HostPort ?? 0;
+
+            if (ip == default || siloPort == default || gatewayPort == default)
+            {
+                throw new Exception($"ECS metadata retrieval failed. Values received: Ip='{ip}', SiloPort='{siloPort}', GatewayPort='{gatewayPort}'.");
+            }
+
+            if (IPAddress.TryParse(ip, out var parsedIp))
```

Comment:
Nah, let's not `Try`. If this thing fails, I want it to fail spectacularly making it immediately obvious what's happening. This would probably not stop the silo from starting and it would just be limping and spamming splunk for a while. Let it fail.